synapse/util/logcontext.py

```diff
@@ -544,6 +545,9 @@ def run_in_background(f, *args, **kwargs):
         # by synchronous exceptions, so let's turn them into Failures.
         return defer.fail()
 
+    if isinstance(res, types.CoroutineType):
```

Comment:
Think I'd like to see some tests that logcontexts are correctly preserved when a coroutine is returned; however I'm happy if you want to punt this to a separate PR provided you are confident that it is handled correctly and have tested it manually or something.

Reply:
my logs didn't exhibit the lost context logging, so I think I fixed it?